src/tree/v2/BranchDataProviderItem.ts

```diff
@@ -72,6 +72,19 @@ export class BranchDataProviderItem implements ResourceGroupsItem, WrappedResour
         };
     }
 
+    async getParent(): Promise<ResourceGroupsItem | undefined> {
+        if (this.branchDataProvider.getParent) {
+            const branchItem = await this.branchDataProvider.getParent(this);
```

Comment:
I'm not sure I follow here; the BDP is expecting one of its own model types, but we're passing it a `BranchDataProviderItem` which is a RG type?

Reply:
I think this was a mistake. I changed it so that we're passing the branch item.